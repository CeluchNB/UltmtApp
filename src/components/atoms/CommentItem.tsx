import { Comment } from '../../types/action'
import { IconButton } from 'react-native-paper'
import React from 'react'
import { size } from '../../theme/fonts'
import { useColors } from '../../hooks'
import { StyleSheet, Text, View } from 'react-native'

interface CommentInputProps {
    comment: Comment
    userId: string
    onDelete: (number: number) => void
}

const CommentItem: React.FC<CommentInputProps> = ({
    comment,
    userId,
    onDelete,
}) => {
    const { user, comment: commentText } = comment
    const { colors } = useColors()

    const styles = StyleSheet.create({
        container: {
            backgroundColor: colors.darkPrimary,
            padding: 20,
            display: 'flex',
            flexDirection: 'row',
        },
        commentContainer: {
            alignItems: 'flex-start',
            flex: 1,
        },
        name: {
            fontSize: size.fontFifteen,
            color: colors.gray,
        },
        comment: {
            fontSize: size.fontFifteen,
            color: colors.textPrimary,
        },
    })
    return (
        <View style={styles.container}>
            <View style={styles.commentContainer}>
                <Text style={styles.name}>
                    {user.firstName} {user.lastName}
                </Text>
                <Text style={styles.comment}>{commentText}</Text>
            </View>
            {userId.length > 0 && userId === comment.user._id && (
                <IconButton
                    icon="close"
                    iconColor={colors.error}
                    onPress={() => {
                        onDelete(comment.commentNumber)
                    }}
                />
            )}
        </View>
    )
}

export default CommentItem
