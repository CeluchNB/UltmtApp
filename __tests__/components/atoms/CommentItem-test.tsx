import CommentItem from '../../../src/components/atoms/CommentItem'
import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'

describe('CommentItem', () => {
    const props = {
        comment: {
            commentNumber: 1,
            comment: 'Test comment',
            user: {
                _id: '',
                firstName: 'First1',
                lastName: 'Last1',
                username: 'firstlast1',
            },
        },
        userId: 'user1',
        onDelete: jest.fn(),
    }

    it('should match snapshot', () => {
        const snapshot = render(<CommentItem {...props} />)
        expect(snapshot).toMatchSnapshot()
    })

    it('should handle delete', () => {
        const { getByRole } = render(
            <CommentItem
                {...props}
                comment={{
                    commentNumber: 1,
                    comment: 'Test comment',
                    user: {
                        _id: 'user1',
                        firstName: 'First1',
                        lastName: 'Last1',
                        username: 'firstlast1',
                    },
                }}
            />,
        )

        const button = getByRole('button')
        expect(button).toBeTruthy()

        fireEvent.press(button)
        expect(props.onDelete).toHaveBeenCalledWith(props.comment.commentNumber)
    })
})
