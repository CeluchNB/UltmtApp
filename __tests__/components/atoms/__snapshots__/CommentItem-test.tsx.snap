// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentItem should match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#1a1a1a",
      "display": "flex",
      "flexDirection": "row",
      "padding": 20,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "flex-start",
        "flex": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "#ababab",
          "fontSize": 30.821917808219176,
          "textDecorationLine": "underline",
        }
      }
    >
      First1
       
      Last1
    </Text>
    <Text
      style={
        {
          "color": "#ffd600",
          "fontSize": 30.821917808219176,
        }
      }
    >
      Test comment
    </Text>
  </View>
</View>
`;
